GVFS/GVFS/CommandLine/SparseVerb.cs

```diff
@@ -88,26 +91,27 @@ protected override void Execute(GVFSEnlistment enlistment)
                         if (directories.Contains(folder))
                         {
                             needToChangeProjection = true;
-                            break;
+                            directories.Remove(folder);
                         }
                     }
 
-                    if (!needToChangeProjection)
+                    foreach (string folder in foldersToAdd)
                     {
-                        foreach (string folder in foldersToAdd)
+                        if (!directories.Contains(folder))
                         {
-                            if (!directories.Contains(folder))
-                            {
-                                needToChangeProjection = true;
-                                break;
-                            }
+                            needToChangeProjection = true;
+                            directories.Add(folder);
                         }
                     }
 
                     if (needToChangeProjection)
                     {
                         // Make sure there is a clean git status before allowing sparse set to change
-                        this.CheckGitStatus(tracer, enlistment);
+                        if (directories.Count > 0)

```

Comment:
Why check `directories.Count > 0` before calling `CheckGitStatus`?  If `directories` is empty does that mean nothing is projected or everything is projected?

Reply:
That means that everything will be projected so there isn't any use checking status because the projection change will not affect any of the dirty files no matter where they are.